### PR TITLE
JS. Stateful clones. Solution

### PR DESCRIPTION
**Pull request by khviktoriia**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,43 @@
...
+    stateCopies.push(stateCopy);
+  }
+
+  stateCopies.shift();
+
+  return stateCopies;
+}
+
+function transform(state, action) {
+  switch (action.type) {
```
Comment (volodymyr-soltys97) [inline src/transformStateWithClones.js:25]: Use a destructuring for `action`

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,43 @@
...
+    case 'addProperties':
+      Object.assign(state, action.extraData);
+      break;
+
+    case 'removeProperties':
```
Comment (volodymyr-soltys97) [inline src/transformStateWithClones.js:30]: Create a variable for these magic words and use them everywhere

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Great work 👍 
Let's improve your code

**Review by anastasiia-tilikina — APPROVED** [review]
Well done!

**Pull request closed without merging**